tests/modules/users/resources/test_options.py

```diff
@@ -0,0 +1,30 @@
+# pylint: disable=missing-docstring
+import pytest
+
+@pytest.mark.parametrize('path,status_code,result', (
+    ('/api/v1/users/', 204, ['POST', 'OPTIONS']),
+    ('/api/v1/users/1', 401, None),
+))
+def test_users_options_unauthorized(path, status_code, result, flask_app_client):
+    response = flask_app_client.options(path)
+
+    assert response.status_code == status_code
+    if result:
+        assert len(
+                list(set(result) - set(response.headers['Allow'].replace(' ', '').split(',')))
```

Comment:
Just for your info, there is no point in converting `set` to `list` since `len()` works just fine on sets.

Also, there is no point in doing `len(...) == 0`, it is just less pythonic, though I think you know. You may also learn about `any` and `all`.

Also, `split()` can handle a string as a parameter, which it will handle as a list of possible separators (not like one single multicharacter separator), so

```python
>>> 'A, B'.split(', ')
['A', 'B']
```

Ultimately, I would write this check like this:

```python
assert not (set(result) - set(response.headers['Allow'].split(', ')))
```